core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -863,6 +872,39 @@ private[spark] class TaskSchedulerImpl(
       executorUpdates)
   }
 
+ private def getTaskAccumulableInfosAndProcessRate(
+     updates: Seq[AccumulatorV2[_, _]]): (Seq[AccumulableInfo], Double) = {
+   var recordsRead = 0L
+   var executorRunTime = 0L
+   val accInfos = updates.map { acc =>
+     if (efficientTaskCalcualtionEnabled && acc.name.isDefined) {
+       val name = acc.name.get
+       if (name == shuffleRead.RECORDS_READ || name == input.RECORDS_READ) {
```

Comment:
Hi @weixiuli , I'm reading through this part of code, and I think it will never work because task metric `EXECUTOR_RUN_TIME` is only set at the end of task. When task is still running, this metric in heartbeat is always zero.  (I might be wrong...please feel free to correct me